File: netgear/RAX29/webs/php/vlan.php
<?
include_once 'common_utils.php';

function getCheckboxVal_enableVlan()
{   
    $enable = db_get("Device.X_PEGATRON_COM_VLAN.Enable");
    
    return num_to_TrueFalseStr($enable);
}

function getRadioVal_groupType()
{
    return db_get("Device.X_PEGATRON_COM_VLAN.Type");
    //return "port";
    //return "vid";
}

function getCheckboxVal_port1()
{
    $portMap = db_get("Device.X_PEGATRON_COM_VLAN.PortMaps");
    $port_array = explode(",", $portMap);
    
    foreach($port_array as $port) {
        if ($port == "port1") {
            return "true";
        }
    }
    
    return "false";
}

function getCheckboxVal_port2()
{
$portMap = db_get("Device.X_PEGATRON_COM_VLAN.PortMaps");
    $port_array = explode(",", $portMap);
    
    foreach($port_array as $port) {
        if ($port == "port2") {
            return "true";
        }
    }
    
    return "false";
}

function getCheckboxVal_port3()
{
    $portMap = db_get("Device.X_PEGATRON_COM_VLAN.PortMaps");
    $port_array = explode(",", $portMap);
    
    foreach($port_array as $port) {
        if ($port == "port3") {
            return "true";
        }
    }
    
    return "false";
}

function getCheckboxVal_port4()
{
    $portMap = db_get("Device.X_PEGATRON_COM_VLAN.PortMaps");
    $port_array = explode(",", $portMap);
    
    foreach($port_array as $port) {
        if ($port == "port4") {
            return "true";
        }
    }
    
    return "false";
}

function getCheckboxVal_wifi2g()
{
    $portMap = db_get("Device.X_PEGATRON_COM_VLAN.PortMaps");
    $port_array = explode(",", $portMap);
    
    foreach($port_array as $port) {
        if ($port == "wifi2g") {
            return "true";
        }
    }
    
    return "false";
}

function getCheckboxVal_wifi5g()
{
    $portMap = db_get("Device.X_PEGATRON_COM_VLAN.PortMaps");
    $port_array = explode(",", $portMap);
    
    foreach($port_array as $port) {
        if ($port == "wifi5g") {
            return "true";
        }
    }
    
    return "false";
}

function getVlanRuleTable()
{
    $db_path = "Device.X_PEGATRON_COM_VLAN.Group";
    $json = db_getObj($db_path);
    if($json === FALSE)
    {
        return ;  
    }

    $data = json_decode($json,true);
    if (json_last_error() != JSON_ERROR_NONE) {
        printf("JSON Error: %s", json_last_error_msg());
    }
    
    $output_value = new stdClass();
    $count = 1;
    
    foreach($data[$db_path] as $inst)
    {
        //get index    
        $index = $inst["instanceID"];
            
        //get enable
        if ($inst["Enable"] == "1") {
            $enable = "checked";
        }
        else if ($inst["Enable"] == "0") {
            $enable = "";
        }
    
        //get port 
        $portMap = $inst["PortMaps"];
        $port_array = explode(",", $portMap);
        $port1 = "";
        $port2 = "";
        $port3 = "";
        $port4 = "";
        $wifi2g = "";
        $wifi5g = "";
    
        foreach($port_array as $port) {
            if ($port == "port1") {
                $port1 = "checked";
            }
            if ($port == "port2") {
                $port2 = "checked";
            }
            if ($port == "port3") {
                $port3 = "checked";
            }
            if ($port == "port4") {
                $port4 = "checked";
            }
            if ($port == "wifi2g") {
                $wifi2g = "checked";
            }
            if ($port == "wifi5g") {
                $wifi5g = "checked";
            }
        }
            
        
        if ($inst["Display"] == "0") {
            $display = 'style="display:none;"';
        }
        else {
            $display = "";
        }
            
            $output_value->{$count} = (object) null;
        $output_value->{$count}->__index = $index;
        $output_value->{$count}->__enableRule = $enable;
        $output_value->{$count}->__name = htmlentities($inst["GroupName"]);
        $output_value->{$count}->__escapedName = escapeBackslashSinglequote($output_value->{$count}->__name);
        $output_value->{$count}->__vlandId = $inst["Vid"];
        $output_value->{$count}->__priority = $inst["Priority"];
        $output_value->{$count}->__enPort1 = $port1;
        $output_value->{$count}->__enPort2 = $port2;
        $output_value->{$count}->__enPort3 = $port3;
        $output_value->{$count}->__enPort4 = $port4;
        $output_value->{$count}->__enWifi2g = $wifi2g;
        $output_value->{$count}->__enWifi5g = $wifi5g;
        $output_value->{$count}->__displayStyle = $display;
        $output_value->{$count}->__displayVal = $inst["Display"];
        
            $count++;
    }
    return $output_value;
}
?>

File: netgear/RAX5/usr/lib/lua/webGetFunc/vlan.lua
local M = {}
local uci  = require "luci.model.uci".cursor()
local json = require "luci.json"
local log = require "luci.log" -- for debug
local util = require "luci.util"

function M.getCheckboxVal_enableVlan()
    local enableVlan = uci:get("vlan", "@VlanCfg[0]", "enableVlan")
    return enableVlan
end

function M.getRadioVal_groupType()
    local group_type = uci:get("vlan", "@VlanCfg[0]", "groupType")
    return group_type
end

function M.getCheckboxVal_port1()
    local port1 = uci:get("vlan", "BridgeGP", "port1")
    return port1
end
function M.getCheckboxVal_port2()
    local port2 = uci:get("vlan", "BridgeGP", "port2")
    return port2
end
function M.getCheckboxVal_port3()
    local port3 = uci:get("vlan", "BridgeGP", "port3")
    return port3
end
function M.getCheckboxVal_port4()
    local port4 = uci:get("vlan", "BridgeGP", "port4")
    return port4
end
function M.getCheckboxVal_wifi2g()
    local wifi2g = uci:get("vlan", "BridgeGP", "wifi2g")
    return wifi2g
end
function M.getCheckboxVal_wifi5g()
    local wifi5g = uci:get("vlan", "BridgeGP", "wifi5g")
    return wifi5g
end

function M.getVlanRuleTable()
    log.debug(0)
    local vlan_rule_table = {}
    local index = 1

    uci:foreach("vlan", "switch_vlan",
      function(s)
      --log.console_r(s)

      if s.name == 'EtherWan' then return true end
      if s.name == 'Bridge' then return true end

      local chk_port1 = ''
      local chk_port2 = ''
      local chk_port3 = ''
      local chk_port4 = ''
      if s.ports ~= nil then
        port_list = util.split(s.ports," ")

        for i in ipairs(port_list) do
          --log.console(i, port_list[i])
          if port_list[i] == '1' then
            chk_port1 = "checked"
          elseif port_list[i] == '2' then
            chk_port2 = "checked"
          elseif port_list[i] == '3' then
            chk_port3 = "checked"
          elseif port_list[i] == '4' then
            chk_port4 = "checked"
          end
        end
      end

      local wifi2g_ck, wifi5g_ck=''
      if s.name == 'Internet' then
	vid = uci:get("vlan", "EtherWan", "vid")
        ig_wifi2g = uci:get("vlan", "Internet", "wifi2g")
        ig_wifi5g = uci:get("vlan", "Internet", "wifi5g")
        if ig_wifi2g=='true' then wifi2g_ck="checked" end
        if ig_wifi5g=='true' then wifi5g_ck = "checked" end
      else
	vid = s.vid
        if s.wifi2g=='true' then wifi2g_ck="checked" end
        if s.wifi5g=='true' then wifi5g_ck="checked" end
      end

      if s.enable_rule == '1' then
         chk_enRule ="checked"
      else
         chk_enRule=""
      end
      vlan_rule_table[tostring(index)] = {
         __index = index,
         __iid = index,
         __name = s.name,
         __enableRule= chk_enRule,
         __vlandId = vid,
         __priority = s.priority,
         __enPort1 = chk_port1,
         __enPort2 = chk_port2,
         __enPort3 = chk_port3,
         __enPort4 = chk_port4,
         __enWifi2g = wifi2g_ck,
         __enWifi5g = wifi5g_ck
      }
      index = index + 1
    end)

    return vlan_rule_table
end

return M
